### DISCUSSION TITLE
SSO with Nextcloud

### DISCUSSION THREAD

**Discussion by erichrueger** [gary-kim/riotchat#311] [Q&A]
Hi,

no matter what I try I can't get SSO running with nextcloud. Is there a guide what to put in as oidc_provider?
I followed a guide from here:
https://github.com/matrix-org/synapse/issues/8511
Have I misconfigured the oauth section in nextcloud? I don't know what to put in as redirection-uri
I tried the following:
https://matrix.example.de/_synapse/client/oidc/callback
After that, nextcloud displays the dialogue for granting access. But then I get the following error message:
_There was an error 
localpart is invalid:
Error code
mapping_error_

The synapse server displays the following error message:
_File "/usr/local/lib/python3.8/site-packages/synapse/handlers/sso.py", line 606, in _register_mapped_user
    raise MappingException("localpart is invalid: %s" % (attributes.localpart,))_

Can anyone help?
Best regards

Erich

**Comment by jonathanmmm** [comment]
This isn't a webapp problem, more a synapse or reverse proxy problem, do you use a reverse proxy?

If yes, follow the following to get it right:
https://matrix-org.github.io/synapse/develop/reverse_proxy.html

and here to get synapse homeserver.yaml and right
https://matrix-org.github.io/synapse/develop/openid.html

or both under here: https://github.com/matrix-org/synapse/blob/develop/docs/

**Comment by gymnae** [comment]
This made it work for me, but I'm still having issues with the CSRF when initially signing up:
https://github.com/matrix-org/synapse/issues/8511#issuecomment-1527159422
